src/main/java/frc/robot/commands/DefaultMechCommand.java

```diff
@@ -0,0 +1,202 @@
+
+        mReadyForIntake = new ReadyForIntake(intakeSubsystem, launcherSubsystem);
+        mIntaken = new Intaken(launcherSubsystem);
+        mReadyForHandoff = new ReadyForHandoff(intakeSubsystem, launcherSubsystem);
+        mReadyForAim = new ReadyForAim(launcherSubsystem, aimingSubsystem);
+        mReadyForLaunch = new ReadyForLaunch(launcherSubsystem);
+
+        mMechState = determineState();
+    }
+
+    public MechState determineState() {
+        if (getIntakeBeamBreak() && getIndexerBeamBreak()) {
+            return mReadyForIntake;
+        }
+        else if (!getIntakeBeamBreak() && getIndexerBeamBreak()) {
+            return mIntaken;
+        }
+        else if (!getIntakeBeamBreak() && getIndexerBeamBreak() && mAimState == AimState.STOW) {
+            return mReadyForHandoff;
+        }
+        else if (!getIndexerBeamBreak()) {
+            return mReadyForAim;
+        }
+        else if (getIndexerBeamBreak() && isFlywheelAtSP() && isAimAtSP() && isVisionAligned() && Input.getRightBumper()) {
+            return mReadyForLaunch;
+        }
+        return null;
+    }
+
+    @Override
+    public void execute() {
+        if (Input.getX()) {
+            mMechState.setFlywheelOff();
+        }
+        else if (Input.getY()) {
+            mMechState.setSpeakerSP();
+            if (mMechState != mReadyForLaunch) {
+                mLauncherMode = LauncherMode.SPEAKER;
+            }
+            if (mMechState == mReadyForAim) {
+                mAimState = AimState.SPEAKER;
+            }
+        } 
+        else if (Input.getA()) {
+            mMechState.setAmpSP();
+            if (mMechState != mReadyForLaunch) {
+                mLauncherMode = LauncherMode.AMP;
+            }
+            if (mMechState == mReadyForAim) {
+                mAimState = AimState.AMP;
+            }
+        }
+        else if (Input.getB()) {
+            mMechState.setTrapSP();
+            if (mMechState != mReadyForLaunch) {
+                mLauncherMode = LauncherMode.TRAP;
+            }
+            if (mMechState == mReadyForAim) {
+                mAimState = AimState.CLIMB;
+            }
+        }
+        if (Input.getLeftBumper()) {
+            mMechState.intake();
+        }
+        if (Input.getRightBumper()) {
+            mMechState.launch();
+        }
+        if (Math.abs(Input.getLeftStickY()) > 0) {
+            mMechState.controlWrist();
+        }
+        if (Math.abs(Input.getRightStickY()) > 0) {
+            mMechState.controlElevator();
+        }
+        if (Input.getDPad() == Input.DPADUP) {
+            mMechState.setElevatorSPtoStage();
+        }
+        else if (Input.getDPad() == Input.DPADDOWN) {
+            mMechState.setElevatorSPtoBase();
+        }
+        if (Input.getDPad() == Input.DPADLEFT) {
+            mMechState.resetEncoders();
+        }
+
+        runAction();
+
+        mMechState = determineState();
+    }
+
+    //automatic actions
+    public void runAction() {
```

Comment:
The whole point of creating the interfaces was so that you could simply call mMechState.--insert whatever command-- (). Do that event handling within the MechState classes themselves